planner/core/logical_plan_builder.go

```diff
@@ -6763,7 +6763,22 @@ func (b *PlanBuilder) buildWindowFunctionFrameBound(_ context.Context, spec *ast
 	if err != nil {
 		return nil, err
 	}
-	bound.CmpFuncs[0] = expression.GetCmpFunction(b.ctx, orderByItems[0].Col, bound.CalcFuncs[0])
+
+	cmpDataType := expression.GetAccurateCmpType(orderByItems[0].Col, bound.CalcFuncs[0])
+
+	switch cmpDataType {
+	case types.ETInt:
+		bound.CmpFuncs[0] = expression.CompareInt
+		bound.CmpDataType = tipb.RangeCmpDataType_Int
+	case types.ETReal:
+		bound.CmpFuncs[0] = expression.CompareReal
+		bound.CmpDataType = tipb.RangeCmpDataType_Float
+	case types.ETDecimal:
+		bound.CmpFuncs[0] = expression.CompareDecimal
+		bound.CmpDataType = tipb.RangeCmpDataType_Decimal
+	default:
+		return nil, expression.ErrIncorrectType.GenWithStackByArgs("Invalid comparison data type for range frame")
```

Comment:
The implementation seems not the same as before? Before this pr there is `ETDatetime/ETTimestamp`?

Reply:
> The implementation seems not the same as before? Before this pr there is `ETDatetime/ETTimestamp`?

The cmpType here is impossible to be `ETDatetime/ETTimestamp` as `cmpDataType := expression.GetAccurateCmpType(orderByItems[0].Col, bound.CalcFuncs[0])` won't return `ETDatetime/ETTimestamp` in window function